File: easy/225_testing.c
#include <stdio.h>
#include <string.h>

#define LINE_LEN    1024

char *bug_count_to_priority(int bug_count) {
    if (bug_count == 0)
        return "Done";
    if (bug_count <= 2)
        return "Low";
    if (bug_count <= 4)
        return "Medium";
    if (bug_count <= 6)
        return "High";
    return "Critical";
}

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[LINE_LEN];

    while (fgets(line, LINE_LEN, file)) {
        if (line[strlen(line)-1] == '\n')
            line[strlen(line)-1] = '\0';

        char *dev_string = line;
        char *p = strstr(line, " | ");
        *p = '\0';
        char *design_string = p + 3;

        int bug_count = 0;
        for (int i=0; i<strlen(dev_string); i++) {
            bug_count += (dev_string[i] != design_string[i]);
        }

        printf("%s\n", bug_count_to_priority(bug_count));
    }

    return 0;
}

File: easy/225_testing.py
import sys

priorities = {
    0: 'Done',
    1: 'Low',
    2: 'Low',
    3: 'Medium',
    4: 'Medium',
    5: 'High',
    6: 'High'
}

for line in sys.stdin:
    dev_string, design_string = line.strip().split(' | ')
    bug_count = sum(a!=b for a,b in zip(dev_string, design_string))
    print(priorities.get(bug_count, 'Critical'))
